### PR TITLE
Fixes #142 Configure golang-ci and run it in Travis builds

### LINKED ISSUE #142: Code quality for 0.27.0 release
**Describe the bug**
Some functions (e.g. `RemoteWallet`'s) are value receivers . 

**Actual behavior**
This implies copy semantics are activated in place .

**Expected behavior**
No copy should be made .

**Possible implementation**
- [x] Transform value receivers into pointer receivers
- [x] Add docstring comments for public functions and value/pointer receivers
- [x] Configure `golang-ci` and run it in Travis builds

### PR DESCRIPTION
**Pull request by m4yk3ldev**
Fixes #142 

Changes:
- Configure `golang-ci` and run it in Travis builds

Does this change need to mentioned in CHANGELOG.md?
yes

### CODE REVIEW

**Comment by olemis** [comment]
@stdevHan will review asap

**Review by olemis — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  # list of build tags, all linters use it. Default is empty list.
+  build-tags:
+
+  # which dirs to skip: they won't be analyzed;
+  # can use regexp here: generated.*, regexp is applied on full path;
+  # default value is empty list, but next dirs are always skipped independently
+  # from this option's value:
+  #   	vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
+  skip-dirs:
+    - /usr/local/go/src
```
Comment (stdevAlDen) [inline .golangci.yml:27]: @stdevHan is possible to use a env var here?

File: .travis.yml [hunk 2]
```diff
@@ -43,6 +43,7 @@ matrix:
     - make install-deps 
     
     script:
+    - make lint
```
Comment (olemis) [inline .travis.yml:46]: Run on all platforms , just be sure linter will work on all platforms.

**Review by olemis — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 1 above)
Comment (m4yk3ldev) [reply .golangci.yml:27]: I have not tried it before, I will review

**Review by stdevMac — APPROVED** [review]

**Pull request merged**